### PR TITLE
Attempt to migrate to PD API v2

### PR DESCRIPTION
**Pull request by quiffman**
This is a work-in-progress/request-for-comment.

Basically, I need help.

### CODE REVIEW

File: cmd/pd-list-incidents/pd-list-incidents.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	if err := json.Unmarshal([]byte(filter), &opts); err != nil {
+		log.Fatalln("Failed to parse filter: " + err.Error())
+	}
+
+	log.Printf("opts: %v\n", opts)
+	resp, err := pd.ListIncidents(opts)
+
+	if err != nil {
+		log.Fatalln("Failed to fetch incidents for given filter: " + err.Error())
+	} else {
```
Comment (ozym) [inline cmd/pd-list-incidents/pd-list-incidents.go]: Only cosmetic, but the "else" isn't needed here due to the "fatal" return.

e.g:
```go
resp, err := pd.ListIncidents(opts)
 if err != nil {
	log.Fatalln("Failed to fetch incidents for given filter: " + err.Error())
}

for _, i := range resp.Incidents {
```

File: cmd/pd-reassign-all/pd-reassign-all.go [hunk 2]
```diff
@@ -50,40 +48,52 @@ func init() {
...
-		rOpts.EscalationLevel = toLevel
+		manageIncident.EscalationLevel = uint(toLevel)
 	}
 
-	incidents, err := pd.Incidents.ListAll(&pagerduty.IncidentsOptions{AssignedToUser: u.ID})
+	resp, err := pd.ListIncidents(pagerduty.ListIncidentsOptions{UserIDs: []string{u.ID}})
 
 	if err != nil {
 		log.Fatalln("Failed to fetch incidents for user: " + err.Error())
 	} else {
```
Comment (ozym) [inline cmd/pd-reassign-all/pd-reassign-all.go]: As per above re "else"

**Review by ozym — COMMENTED** [review]
Only a quick look through of the code, testing the service (and responses) would need a little more bits. Only spotted some cosmetic issues.

**Comment by quiffman** [comment]
Thanks @ozym 

You're right about the testing.
I guess this is function is a convenience rather than critical.

**Pull request merged**